src/dotnet/ShellShim/WindowsEnvironmentPath.cs

```diff
@@ -33,9 +33,14 @@ public void AddPackageExecutablePathToUserPath()
 
             var existingUserEnvPath = Environment.GetEnvironmentVariable(PathName, EnvironmentVariableTarget.User);
 
+            if (existingUserEnvPath.EndsWith(';'))
+            {
+                existingUserEnvPath = existingUserEnvPath.Substring(0, (existingUserEnvPath.Length - 1));
+            }
+
             Environment.SetEnvironmentVariable(
                 PathName,
-                $"{existingUserEnvPath};{_packageExecutablePath}",
+                $"{existingUserEnvPath};{_packageExecutablePath};",
```

Comment:
Where did you get the information that final ; is required? It's not set on my Windows machine.

Reply:
https://stackoverflow.com/questions/11391390/placing-the-semicolon-in-the-windows-path-environment-variable

Reply:
in a clean vm, there is a ; at the end

Reply:
That link doesn't say it's required nor is it normative. 

It's extremely common to not have a ; at the end.

Does clean VM have empty user PATH? Does it remain that way after you add to user path?

Reply:
I think the earlier line being defensive against ; at the end already there is good, but this part is unnecessary.

Reply:
> It's extremely common to not have a ; at the end. 

Agree. Although the code checks that.

> Does clean VM have empty user PATH? Does it remain that way after you add to user path?

No, the empty PATH is due to no check on the extra ; at the end. ";;" after split causes "" in the array of paths. Basically CLI's non perfect way adding path causes problem for CLI's non perfect way reading the path.

It is fine for Windows to read it. Since Windows will skip ";;"

Reply:
I understand the where the empty PATH was from. As I said, I think the fix a few lines above is good.

I am only wondering if we really need to add this ; on this line at the end. It seems like it only makes ;; more likely for someone else to trip on, and I can find nothing that says it's required.

My question was trying to figure out when a ; at the end of PATH goes away on a clean VM. I never recall seeing it that way before. I was just guessing that once user PATH is set then there is no semicolon at the end.